src/test/java/com/haymetric/metrics/DimensionUtilsTest.java

```
@@ -0,0 +1,61 @@
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions
+ * of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+ * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+package com.haymetric.metrics;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DimensionUtilsTest {
+
+    @Test
+    public void testParseDimensions() {
+        final Map<String, String> expectedOutput = new HashMap<String, String>(){{
+            put("key1", "Tada");
+        }};
+        Assert.assertTrue(DimensionsUtils.parseDimensions("").isEmpty());
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" somekey ").isEmpty());
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" somekey = , somekey2 = ").isEmpty());
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" somekey, somekey2, somekey3 ").isEmpty());
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" kEY1 = Coool, KEY1   = Tada   ").equals(expectedOutput));
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" kEY1 = , KEY1   = Tada, HelloWorld   ").equals(expectedOutput));
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" kEY1 = , KEY1   = Tada, HelloWorld   ").equals(expectedOutput));
+        Assert.assertTrue(DimensionsUtils.parseDimensions(" kEY2 = #@, KEY1   = Tada   ").equals(expectedOutput));
```

We can report an error metric like below for every service.
```
d.service=ImageSharing, m.InvalidMetricNamesAndDimensions=101
```